CAKE-83: display CMS editor when editing MW pages

owend opened the pull request:
use hooks to disable core EditPage, load a new one that does nothing.
loadeditor.js just does two things. 1) demonstrates loading the wikia editor framework 2) injects a new page from wgContributionPrototypeExternalHost (that doesn't work yet)  

@gbenson-wikia

Commit:
CAKE-83: first attempt at loading remote editor

WikiaTech commented on extensions/wikia/ContributionPrototype/ContributionPrototype.setup.php:
```diff
@@ -12,5 +12,16 @@
+//This hook just lets us override the PHP edit class during the edit action
+//Not the "right" way to do this but we erase any other AlternateEditPageClass hooks this way because there can be only one
+$wgHooks['AlternateEditPageClass'] = ['ContributionPrototype\ContributionPrototypeHooks::onAlternateEditPageClass'];
+#$wgHooks['AlternateEditPageClass'][] = ['ContributionPrototype\ContributionPrototypeHooks::onAlternateEditPageClass'];
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)

WikiaTech commented on extensions/wikia/ContributionPrototype/CPEditor.php:
```diff
@@ -0,0 +1,27 @@
+
+	use Loggable;
+
+	function __construct(Article $article) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method "__construct" to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1185)

WikiaTech commented on extensions/wikia/ContributionPrototype/CPEditor.php:
```diff
@@ -0,0 +1,27 @@
+	}
+
+	function edit() {
+		global $wgOut, $wgRequest, $wgUser;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgRequest" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgUser" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,31 @@
+		});
+
+		function init(window) {
+			console.log("CP init and do something");
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,31 @@
+		'use strict';
+
+		$(function() {
+			console.log("CP load editor");
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)

WikiaTech commented on extensions/wikia/ContributionPrototype/ContributionPrototypeHooks.php:
```diff
@@ -0,0 +1,45 @@
+	 * @param $user \User
+	 * @return bool
+	 */
+	public static function onCustomEditor( $page, $user ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$user". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

WikiaTech commented:
SonarQube analysis reported 9 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 8 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 1 minor

Watch the comments in this conversation to review them.

WikiaTech commented on extensions/wikia/ContributionPrototype/ContributionPrototypeHooks.php:
```diff
@@ -0,0 +1,45 @@
+	 */
+	public static function onCustomEditor( $page, $user ) {
+		global $wgExtensionsPath;
+		$title = $page->getTitle();
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$title" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on extensions/wikia/ContributionPrototype/ContributionPrototypeHooks.php:
```diff
@@ -0,0 +1,45 @@
+	public static function onCustomEditor( $page, $user ) {
+		global $wgExtensionsPath;
+		$title = $page->getTitle();
+		$request = RequestContext::getMain()->getRequest();
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$request" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

gbenson-wikia committed:
loads the article view and kicks off the app as a half-step to running the editor

WikiaTech commented on extensions/wikia/ContributionPrototype/CPEditor.php:
```diff
@@ -0,0 +1,35 @@
+	}
+
+	function edit() {
+		global $wgOut, $wgRequest, $wgUser, $wgContributionPrototypeExternalHost;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgRequest" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgUser" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+//					xhr.setRequestHeader('x-wikia-community', window.wgDBname);
+//				},
+//				success: function(result) {
+//					console.log(result);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+//				},
+//				success: function(result) {
+//					console.log(result);
+//					$("#editarea").html(result);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+
+		});
+
+		function init(window) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "window". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+//				}
+//			});
+//			$.get( url, null, function(result) {
+//				$("#editarea").html(result);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+
+			loadHTML().then(injectHTMLAndLoadVendor).then(loadApp);
+
+//			var url = window.wgContributionPrototypeExternalHost + '/wiki/' + window.wgDBname + '/wiki/' + window.wgTitle;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

WikiaTech commented:
SonarQube analysis reported 15 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 1 minor

Watch the comments in this conversation to review them.

WikiaTech commented on extensions/wikia/ContributionPrototype/scripts/LoadEditor.js:
```diff
@@ -0,0 +1,71 @@
+//				url: window.wgContributionPrototypeExternalHost + '/wiki/' + window.wgTitle,
+//				type: "GET",
+//				beforeSend: function(xhr) {
+//					xhr.setRequestHeader('x-wikia-community', window.wgDBname);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

nmonterroso committed:
Merge remote-tracking branch 'origin/dev' into CAKE-83

WikiaTech commented:
SonarQube analysis reported 15 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 1 minor

Watch the comments in this conversation to review them.

nmonterroso committed:
when viewing the CP change overflow-y so it's scrollable

nmonterroso committed:
use wgActions to overload

nmonterroso committed:
set scripts to load at the bottom of the body tag, else the div with id="app" might not be loaded in time

nmonterroso committed:
Merge pull request #12016 from Wikia/cake-903

Cake 903 - use the new /edit action from the CP when editing an article in MW

WikiaTech commented on extensions/wikia/ContributionPrototype/CPArticleRenderer.php:
```diff
@@ -45,17 +45,17 @@ public function __construct($publicHost, $wikiId, $dbName, $urlProvider) {
+		$content = $this->getArticleContent($title, $action);
 		
 		if ($content === false) {
 			// TODO: what do we want to show here?
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on extensions/wikia/ContributionPrototype/CPArticleRenderer.php:
```diff
@@ -69,15 +69,19 @@ private function addStyles(OutputPage $output) {
 
-	private function getArticleContent($title) {
+	private function getArticleContent($title, $action) {
 //		$internalHost = $this->urlProvider->getUrl(self::SERVICE_NAME);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

WikiaTech commented:
SonarQube analysis reported 2 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 1 major
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 1 info

Watch the comments in this conversation to review them.

drsnyder committed:
In-line the design system sprite for the CP

In-line the design system sprite for the CP article views. Direct sprite
loading is not currently provided by the DS so are leveraging renderSvg
to slurp in the contents of sprite.svg and dump it onto the page.

nmonterroso committed:
Merge remote-tracking branch 'origin/dev' into CAKE-83

nmonterroso committed:
Merge pull request #12028 from Wikia/cake-900-83

CAKE-900 In-line the design system sprite for the contribution prototype embedding

nmonterroso committed:
fixing button heights

nmonterroso committed:
Merge pull request #12029 from Wikia/cake-899

fixing CMS buttons in MW

nmonterroso committed:
set button styling when looking at modals

nmonterroso committed:
setting header styles for editing with cms

nmonterroso committed:
load the right rail when in edit mode and using the CMS editor

WikiaTech commented:
SonarQube analysis reported 24 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 7 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/9a4d8b9901b752796e39b28b600ecc9181545423/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

nmonterroso committed:
passing isEditPage so initial render of right rail is correct

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/ff2dcce348bbe41dbab7c05e46e41a3a348c16e1/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

WikiaTech commented on extensions/wikia/Rail/RailController.class.php:
```diff
@@ -13,6 +13,7 @@ public function executeIndex($params) {
 
 		$railModules = isset($params['railModuleList']) ? $params['railModuleList'] : [];
 
+		$this->isEditPage = isset($params['isEditPage']) ? $params['isEditPage'] : false;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

nmonterroso committed:
fixing some style

nmonterroso committed:
Merge remote-tracking branch 'origin/dev' into CAKE-83

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/3cbbde99363673477df005714cbc2a1d8119c065/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/d0b2fb0f075a4b356adf56aa67f18f18ab3581fe/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

nmonterroso committed:
adding styles for icons in toolbar since they're kinda busted with a prod build

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/738fe82783a9cb0cdd38bd4d1c55613dca55df10/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

nmonterroso committed:
keep the fixed header and change other fixed elements to compensate

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 14 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L51](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L114](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L114): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L143): The Cyclomatic Complexity of this function "getRailModuleList" is 45 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L143](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L143): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L173): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L173](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L173): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L196](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L196): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L209](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L209): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L219](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L219): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BodyController.class.php#L277](https://github.com/Wikia/app/blob/76b9ec18bc9127ec95175c52ebd6c2c61e49ef38/skins/oasis/modules/BodyController.class.php#L277): The Cyclomatic Complexity of this function "executeIndex" is 42 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

drsnyder commented:
LGTM
